LEAF_Request_Portal/View.php

```diff
@@ -21,88 +21,101 @@ public function __construct($db, $login)
+
+                if ($tmp['description'] == 'Note Added') {
+                    $packet['description'] = 'Note Added: ';
+                } else if ($tmp['description'] == '' && $tmp['actionText'] == '' ) {
+                    $packet['description'] = 'Action';
+                } else if($tmp['stepTitle'] != '' && $tmp['dependencyID'] < 0) {
+                    $packet['description'] = $tmp['stepTitle'] . ': ' . $tmp['actionText'];
+                } else {
+                    $packet['description'] = $tmp['description'] . ': ' . $tmp['actionText'];
+                }
+
+                $packet['comment'] = $tmp['comment'];
+
+                if ($tmp['userID'] != '') {
+                    $user = $dir->lookupLogin($tmp['userID']);
+                    $name = isset($user[0]) ? "{$user[0]['Fname']} {$user[0]['Lname']}" : $tmp['userID'];
+                    $packet['userName'] = $name;
+                }
+
+                $result[] = $packet;
             }
-            $result[] = $packet;
-        }
 
-        $vars = array(':recordID' => $recordID);
-        $res = $this->db->prepared_query('SELECT signatureID, signature, recordID, stepID, dependencyID, userID, timestamp, stepTitle FROM signatures
-                                            LEFT JOIN workflow_steps USING (stepID)
-	    									WHERE recordID=:recordID', $vars);
+            $sql2 = 'SELECT signatureID, signature, recordID, stepID,
+                        dependencyID, userID, timestamp, stepTitle
+                     FROM signatures
+                     LEFT JOIN workflow_steps USING (stepID)
+                     WHERE recordID=:recordID';
 
-        foreach ($res as $tmp)
-        {
-            $packet = [];
-            $packet['time'] = $tmp['timestamp'];
-            $packet['description'] = $tmp['stepTitle'] . ': Digitally Signed';
-            $packet['comment'] = 'Signature Hash: ' . $tmp['signature'];
+            $res = $this->db->prepared_query($sql2, $vars);
 
-            if ($tmp['userID'] != '')
-            {
-                $user = $dir->lookupLogin($tmp['userID']);
-                $name = isset($user[0]) ? "{$user[0]['Fname']} {$user[0]['Lname']}" : $tmp['userID'];
-                $packet['userName'] = $name;
+            foreach ($res as $tmp) {
+                $packet = [];
+                $packet['time'] = $tmp['timestamp'];
+                $packet['description'] = $tmp['stepTitle'] . ': Digitally Signed';
+                $packet['comment'] = 'Signature Hash: ' . $tmp['signature'];
+
+                if ($tmp['userID'] != '') {
+                    $user = $dir->lookupLogin($tmp['userID']);
+                    $name = isset($user[0]) ? "{$user[0]['Fname']} {$user[0]['Lname']}" : $tmp['userID'];
+                    $packet['userName'] = $name;
+                }
+
+                $result[] = $packet;
             }
-            $result[] = $packet;
+
+            usort($result, function($a, $b) {
```

Comment:
Two things.  1) No nested functions.  2) It doesn't look like it even gets called?

Reply:
I wasn't sure what this was even. I just left it because I didn't want to mess anything up.

Reply:
I think this is sorting the result array. I pulled it out into a separate static function